### DISCUSSION TITLE
Curating mapsets metadata to be on par with today's standards

### DISCUSSION THREAD

**Discussion by Skubers** [ppy/osu#22376] [Ideas]
I'll start with a disclaimer that I'm not witty in terms of things being 'easy/possible' to develop technologically, so if something is definitely out of scope, don't come at me for not knowing. If it was dependent on me, I would make a whole app on If-Then statements lol.

I've thought of that last night, when trying to bore myself to sleep.

**IDEA:**
Having a group of 'Curators' check all mapsets from 2007-2018 (and later if needed), and 'curate' the metadata, so that it is correct to today's standards

**WHY?**
This way, ranked mapsets that predate the (Cut Ver.) times, can get actual correct metadata. Not only ones that are missing (Cut Ver.) but also all the oldies, like https://osu.ppy.sh/beatmapsets/7930#osu/37433 with Artist/Source field having 'Nightcore' lol

Additionally, it would fix any post-ranked issues with metadata.

**HOW?**
The idea is to create a couple of new 'variables' that would be entirely web-stored (like Genre and Language). 
These variables would be:
- Curated Artist
- Curated Title
- Curated Source
- Curated Tags

Since it would be web-stored, you don't need to edit the beatmap's file (that way, the Ranked status can be preserved), and for sort/filter purposes, you can always get the data from the web (like with Genre and Language).

There would be a new group of people - Curators, similar to BNs, but their main scope is metadata.
A curator would take a mapset with incorrect metadata, and get the correct metadata for it, submit it (possibly on Mapping Discussions page, or maybe on a page similar to https://bn.mappersguild.com/). To ensure that there are no 'trolls' changing the metadata to wrong stuff, there would be a requirement, so that any submission would have to be reviewed positively by at least 2 other curators.

**EXAMPLE:**
The earlier mentioned Pinball mapset (https://osu.ppy.sh/beatmapsets/7930#osu/37433)

Curator would submit curated metadata for review:
Curated Artist: Cc.K
Curated Title: Pinball (Radio Edit) (Nightcore Mix)
Curated Source:
Curated Tags: CCK C C K Dance Electronic English Trance Hands Up Club

Two other curators would go through the meta to ensure it's not 'troll', and would accept that.
Now the beatmap page shows the curated metadata instead of the original.

**ADDITIONAL INFO:**
People who prefer original metadata (because maybe they remember the beatmaps by their original metadata), would be able to change the setting similar to "Show beatmap metadata in original language", but "Show original uncurated beatmap metadata".
This would switch between either getting the data from the beatmap file, or from the web-stored curated data.
This would similarly work in-game in lazer.

**Comment by Theighlin** [comment]
I think changing metadata would also invalidate the ranking status. I reckon correction for single mapsets has been asked before, but the conclusion was that it was not worth, because you would lose the leaderboard history of the map (that would itself not change in gameplay).

**Reply by Skubers** [reply]
That would be correct if we were to edit the beatmap file to change the data.

The idea is to not do that and instead to curate the metadata entirely on the webpage, like with the Genre and Language. I'm pretty sure BNs or NATs can do that for already Ranked mapsets.
That way the ranked status can be preserved.

**Comment by h3oCharles** [comment]
very late answer, but i feel like pre-2018 is a bit misleading, as i have been an observer of a metadata change that mismatches an already ranked map

pending map (and discussion around it) being https://osu.ppy.sh/beatmapsets/1843213/discussion/-/generalAll#/3591127/9650223
already ranked map being https://osu.ppy.sh/beatmapsets/1478451#taiko/3033142

I'm sure there can be more than just that

PS. initially i thought of following up with providing a lot of worst offenders, but i lost my attention span. obviously, the only ppl who could do this is NAT/GMT and higher, but even then, i feel like a separate group just dedicated to that might come in handy

**Reply by Skubers** [reply]
Yeah, that makes sense, I'll delete the 'pre-2018' thing

**Comment by h3oCharles** [comment]
here are some of the notable offenders that i found throughout the years:

1. inconsistent romanization
-- https://osu.ppy.sh/beatmapsets?q=Dakara%2C%20Hitori%20ja%20nai%20janai
2. already mentioned, but Nightcore as artist
-- https://osu.ppy.sh/beatmapsets?q=nightcore
3. literally empty metadata fields
-- https://osu.ppy.sh/beatmapsets/2490
4. same metadata, but different arrangements
-- https://osu.ppy.sh/beatmapsets/1236988
-- https://osu.ppy.sh/beatmapsets/622136
5. cut ver missing
-- https://osu.ppy.sh/beatmapsets/765778
-- https://osu.ppy.sh/beatmapsets/873811
6. TV Size marker not normalized
-- https://osu.ppy.sh/beatmapsets/596704
-- https://osu.ppy.sh/beatmapsets/127683
7. marker removed...?
-- https://osu.ppy.sh/beatmapsets/983911 (also cut ver missing)
-- https://osu.ppy.sh/beatmapsets/1227249
8. grammar error on sped up ver.
-- https://osu.ppy.sh/beatmapsets/984684
9. one specific song has two titles...?
-- https://osu.ppy.sh/beatmapsets?q=Asu%20no%20Yozora%20Shoukaihan
10. capitalization error
-- https://osu.ppy.sh/beatmapsets/756794
11. ft. and similar not normalized to feat.
-- https://osu.ppy.sh/beatmapsets/874948
-- https://osu.ppy.sh/beatmapsets/624879
-- https://osu.ppy.sh/beatmapsets?q=ft.
12. evangelion op got different metadata (I'm sure there are more songs than that
-- https://osu.ppy.sh/beatmapsets/1377
-- https://osu.ppy.sh/beatmapsets/1799692
13. some maps don't have romanization
-- https://osu.ppy.sh/beatmapsets/1

ill come back to this later

**Reply by Skubers** [reply]
tbh, no need to list the maps, there are hundreds if not thousands of them, all of which became invalid because of the changes to the metadata ranking criteria

**Comment by aryoadhi** [comment]
Metadata curation is an excellent idea. This resolves the headache where metadata would be _wildly_ different between original and English language, since the curators would've set this to the most correct/official one.

Most notable offenders of this case are Japanese artists, where there are conflicts between standardisation and the preferred written name.

In case of ranking scenario where romanisation vs. translation occurs, mappers will get much more freedom choosing whichever is correct, while keeping it to match ranking criteria regarding metadata. The curated metadata can be a textbook example of metadata needed for a song, sans mapping style.

I would be concerned since some beatmaps have tags that are unrelated to the song, such as beatmapping style or creator's name. This implementation could make it harder for users to search for the beatmap sets they're looking for. Probably we need to implement search filters (e.g search in curated/uncurated/both tags)

Your idea of storing metadata based on curated/original also came to my mind, and I think your proposal explained better than what would be my suggestion if I didn't search it first. Overall, I think this is a great idea that would benefit the osu! community.

**Reply by h3oCharles** [reply]
>such as beatmapping style

if a team of humans would be doing the final say per beatmap/song, this would probs be handled on a case-by-case basis, and/or an ranking criteria update clarifying this

>or creator's name

guest difficulty makers, hitsounders, storyboarders, and old names have their names in tags for searchability per beatmap

**Comment by ReiFan49** [comment]
Curated metadata? Yes please. However there are some differences from my idea with yours.

To me, curated metadata has a place on their own. Nominators and Curators link those curated metadata into a mapset (tho Mapset Owner can do it as well to make sure). Also by this, I would like to see on how "Curated Suffixes" go. Because it's a list of flags that attached to the beatmap title. Let's say "Game Ver.", "Short Ver.", "Cut Ver.", "Full Ver.", "Nighcore Mix", "Sped Up Ver.", etc.

Otherwise, this is an idea, I've been thinking as well for months, that I haven't got the time to execute it. Kudos for you man.

**Reply by h3oCharles** [reply]
probably conflicts with [#5852](https://github.com/ppy/osu-web/issues/5852)

**Comment by h3oCharles** [comment]
came back to this again, and here are my current thoughts:
- with my previous attempts of proposing more user groups failed when NAT/GMT can do the same tasks, I'm unsure if adding one for this is necessary
- unsure how DMCA'd maps would be handled
- should the work of metadata curation be shown first to evaluate the need for this?
- should these people be also responsible for adjusting the preview point, as well as adding background images and videos? a lot of the early maps do not have these
- probably conflicts with [#5852](https://github.com/ppy/osu-web/issues/5852) (the proposal of booru-styled tagging system)
- what about songs that make their debut in osu! first, before getting released to music stores?
- should normal people be able to report incorrect metadata with the reporting system?

has this discussion matured enough to warrant an issue on osu-web?

**Reply by Skubers** [reply]
- I think NAT/GMT have enough stuff to do now, no need to give them more work. Also there are groups of people strictly dedicated for finding good meta, so I don't see why we wouldn't discuss that with them.
- Why does DMCA make a difference? This is purely about meta, not about the mp3. Meta isn't changed if a set is DMCA'd.
- BNs and NATs are already doing this part and there are many issues with metadata brought up during the modding phase. I think that we could add a 'nomination' thing but for Curators so that it indicates that the meta was checked by one.
- Adding background images and videos is not possible without tampering with the .osu file, so that's out of question. We're talking purely metadata here.
Preview points are subjective (and also .osu-dependant), so there's no need to adjust them
- I don't see why this (https://github.com/ppy/osu-web/issues/5852) would be a problem tbh
- I don't think I've ever heard about such a situation outside of Featured Artists with whom osu! staff have close contacts so establishing correct metadata wouldn't be hard
- It could prove useful, it could work the same way as 'reporting a problem with a map' which pings NATs and BNs if the mapset is in Qualified

I think we've tackled many of the 'what-ifs' and 'how-abouts' that we can try go further with this, but I'm not sure how to do that lol

**Reply by h3oCharles** [reply]
i did it https://github.com/ppy/osu-web/issues/10108

**Comment by Skubers** [comment]
So I just noticed that Tags are now editable on osu! website, which were locked behind .osu file up until now. 
![image](https://user-images.githubusercontent.com/82095491/230970522-4a00aa42-e8de-4595-8410-6195b7cee795.png)

This means, that what I've proposed is 100% doable

**Comment by peppy** [comment]
The end goal is that when we reimplement beatmap submission, it will be modular and flexible enough that we can rewrite *all* beatmaps (`osz` files) serverside with up-to-date metadata, populated `OnlineID` etc.

A lot of what you've asked for here should be covered by that process.
